src/core/feedforward.jl

```diff
@@ -1,7 +1,88 @@
+############################Chronologies For FeedForward###################################
+@doc raw"""
+    Synchronize(periods::Int64)
+Defines the co-ordination of time between Two stages.
+
+# Arguments
+- `periods::Int64`: Number of time periods to grab data from
+"""
+struct Synchronize <: FeedForwardChronology
+    periods::Int64
+    function Synchronize(; periods)
+        new(periods)
+    end
+end
+
+"""
+    RecedingHorizon(period::Int64)
+""" # TODO: Add DocString
+struct RecedingHorizon <: FeedForwardChronology
+    period::Int64
+    function RecedingHorizon(; period::Int64 = 1)
+        new(period)
+    end
+end
+
+struct Consecutive <: FeedForwardChronology end
+
+function check_chronology(sim::Simulation, key::Pair, sync::Synchronize)
+    from_stage = get_stage(sim, key.first)
+    to_stage = get_stage(sim, key.second)
+    from_stage_horizon = sim.sequence.horizons[key.first]
+    to_stage_horizon = sim.sequence.horizons[key.second]
+    from_stage_interval = sim.sequence.intervals[key.first]
+    to_stage_interval = sim.sequence.intervals[key.second]
+
+    from_stage_resolution =
+        IS.time_period_conversion(PSY.get_forecasts_resolution(from_stage.sys))
+    @debug from_stage_resolution
+    @debug to_stage_interval
+    to_stage_sync = Int(from_stage_resolution / to_stage_interval)
+    from_stage_sync = sync.periods
+
+    if from_stage_sync > from_stage_horizon
+        throw(IS.ConflictingInputsError("The lookahead length $(from_stage_horizon) in stage is insufficient to syncronize with $(from_stage_sync) feedforward periods"))
+    end
+
+    if (from_stage_horizon % from_stage_sync) != 0
+        throw(IS.ConflictingInputsError("The number of feedforward periods $(from_stage_horizon) in stage
+               needs to be a mutiple of the horizon length $(from_stage_horizon)
+               of stage to use Synchronize with parameters ($(from_stage_sync), $(to_stage_sync))"))
+    end
+
+    return
+end
+
+function check_chronology(sim::Simulation, key::Pair, ::Consecutive)
+    from_stage_horizon = sim.sequence.horizons[key.first]
+    from_stage_interval = sim.sequence.intervals[key.first]
+    if from_stage_horizon != from_stage_interval
+        @warn("Consecutive Chronology Requires the same interval and horizon, the parameter horizon = $(from_stage_horizon) in stage $(key.first) will be replaced with $(from_stage_interval). If this is not the desired behviour consider changing your chronology to RecedingHorizon")
+    end
+    sim.sequence.horizons[key.first] = sim.sequence.intervals[key.first]
```

Comment:
This line should be moved inside the if block.  Since you're mutating `sim`, the function should be `check_chronology!`.